src/gd_avif.c

```diff
@@ -0,0 +1,667 @@
+/**
+ * File: AVIF IO
+ *
+ * Read and write AVIF images using libavif (https://github.com/AOMediaCodec/libavif) .
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <limits.h>
+#include <math.h>
+
+#include "gd.h"
+#include "gd_errors.h"
+#include "gdhelpers.h"
+
+#ifdef HAVE_LIBAVIF
+#include "avif.h"
+#include "internal.h"
+// internal.h is only used for AVIF_MIN() and AVIF_CLAMP() macros
```

Comment:
internal.h isn't installed so you cannot use it.  only avif.h is installed.

Reply:
thanks for noticing that. I see it now...

(I was a little nervous about using a file called `internal.h` anyway)

Is it ok if I reproduce these macros? I see them defined in gd_interpolation.c, but not elsewhere in libgd.

Reply:
ugh, what a mess.  rebase onto latest master as i've cleaned things up a little.  if you include gd_intern.h, you'll get MIN/MAX/CLAMP macros.

Reply:
Ok. I'll re-rebase before I keep working on this.

Reply:
Now I see! You just added these macros to gd_intern.h.

I've rebased and I'll happily use your new, globally available, macros.